lib/helper.rb

```diff
@@ -139,6 +139,25 @@ def self.deep_copy(original, override: {}, shallow_copy: [])
     end
   end
 
+  def self.populate_data(services_vrps)
```

Comment:
The `Models::Relation` is never retrieved using its `id`, so it is not mandatory to load it back in memory.

The `split_independent_vrp` is applied on server side, so the worker receive an array of `service_vrp` which indeed share at least their `points`. So it cannot be performed from a single vrp at the moment

Reply:
Relations can be retrieved by their `id` (for example creating or soft-copying a service with an existing relation by id). This has never been done since it wasn't possible since now (and we were used the objects themselves) due to workers' active_hash was missing the id's. But now, that you are adding this functionality in the current PR, we can use `id`s for such task and it would be better to add it for all necessary models, right? Otherwise, we literally don't realise that there is such issue until the code is deployed (or launched via server/worker).

Ideally this load operation would be in base.rb and it would be recursive (i.e., we wouldn't need to load everything manually by hand), I guess, this is out of scope but if you think it is easy to do so, you can even do it more generally like that.

I forgot about the server side :facepalm: cheers. So effectively, doing it *per-VRP* is impossible due to shared objects.

Reply:
Do you expect to have loaded in memory all the sub objects (timewindows, capacities, quantities and locations) ?
By my side I only expect to have the top level objects. For the sub ones, I don't really see the goal of retrieving them from the database as we will always access them within the context of their parent.

Reply:
I think your separation top- vs sub-level is v good. We don't need the subs. My comment about recursive function was wrt ease of maintenance and automatism. 

If we, one day, change the internal model to the "csv" json type in the future (i.e., identical TW/capacity/quantity objects are not recreated to safe memory) then we might need the sub ones but I don't think we need to think about that anytime soon.